### PR TITLE
set_device_template. check ap/os tags

### PR DESCRIPTION
**Pull request by x24ling**
For #337

### CODE REVIEW

**Comment by x24ling** [comment]
Note, i'm will be fixing up the error raising logic, that isn't finalized.

File: iotile_ext_cloud/iotile/cloud/cloud.py [hunk 1]
```diff
@@ -176,6 +175,31 @@ def set_sensorgraph(self, device_id, new_sg):
             else:
                 raise ExternalError("Error calling method on iotile.cloud", exception=exc, response=exc.response.status_code)
 
+    @param("device_id", "integer", desc="ID of the device that we want information about")
+    @param("new_template", "string", desc="The new device template that we want to set")
+    @param("os_tag", "integer", desc="Optional arg to check if the sensorgraph on the cloud matches the os_tag")
+    def set_device_template(self, device_id, new_template, os_tag=None):
+        """Sets the device template for the given device in iotile.cloud."""
+        if os_tag is not None:
+            sg_os_tag = requests.get('https://iotile.cloud/api/v1/dt/moo%s/' % (new_template)).json().get('os_tag', None)
```
Comment (dkarchmer) [inline iotile_ext_cloud/iotile/cloud/cloud.py]: No reason to ever do a manual GET. This is also not checking if the template was found and the cloud had no issues returning the data

You should always use the python_iotile_cloud to do anything on the cloud:

```
try:
     dt = self.api.dt(new_template).get()
except RestHttpBaseException, exc:
     ...
```

under a try/catch just as you do when patching the device

File: iotile_ext_cloud/iotile/cloud/cloud.py [hunk 2]
```diff
@@ -161,9 +161,8 @@ def check_time(self, max_slop=300):
 
     @param("device_id", "integer", desc="ID of the device that we want information about")
     @param("new_sg", "string", desc="The new sensor graph id that we want to load")
-    def set_sensorgraph(self, device_id, new_sg):
+    def set_sensorgraph(self, device_id, new_sg, app_tag=None):
```
Comment (dkarchmer) [inline iotile_ext_cloud/iotile/cloud/cloud.py]: Where are you using this?

You need to do the same

```
try:
     sg = self.api.sg(new_sg).get()
except RestHttpBaseException, exc:
     ...
```

to get the sensor graph and then check that `sg['app_tag'] == app_tag`

**Comment by x24ling** [comment]
To test this function, I also need to break out the mock api calls for sg and dt, I'll be adding that into the mock cloud as well.

**Pull request merged**